src/cosmos/cosmosTools.ts

```diff
@@ -163,6 +168,27 @@ export class CosmosTools implements EdgeCurrencyTools {
     const encodedUri = encodeUriCommon(obj, pluginId, amount)
     return encodedUri
   }
+
+  async connectClient(walletId: string): Promise<TendermintClient> {
+    if (this.client == null) {
+      const stargateClient = await StargateClient.connect(
+        this.networkInfo.rpcNode
+      )
+      // eslint-disable-next-line @typescript-eslint/dot-notation
+      this.client = await stargateClient['forceGetTmClient']()
+    }
+    this.connectedWalletIds.add(walletId)
+    return this.client
+  }
+
+  async disconnectClient(walletId: string): Promise<void> {
+    this.connectedWalletIds.delete(walletId)
+
+    if (this.connectedWalletIds.size === 0) {
+      await this.client?.disconnect()
+      this.client = undefined
+    }
```

Comment:
That's a weird use case! As written it would start to get weird for sure. I can rewrite it to use a counter but I'm going to go farther and redo how the clients are initialized. The implementation swayed back and forth between using multiple types of clients (tendermint, query, stargate, etc) but eventually circled back to just using stargate.

Reply:
This happened to reveal a resync bug